### ISSUE TITLE
Some dependencies not found/shown

### ISSUE THREAD

**Issue by psiinon** [mozilla-services/dependency-observatory#132]
https://depobs.dev.mozaws.net/?manager=npm&package=%40hapi%2Fcode&version=8.0.1 shows no dependencies but https://www.npmjs.com/package/@hapi/code/v/8.0.1 shows it has one (@hapi/hoek)

**Comment by psiinon** [comment]
Looks like the UI is ok - the API call https://depobs.dev.mozaws.net/package?package_name=%40hapi%2Fcode&package_version=8.0.1 reports all_deps = 1 and immediate_deps = 2 (which doesnt make sense) and doesnt return any of the dependency details = dependencies = []

**Comment by psiinon** [comment]
Querying the db directly:
```
SELECT pv2.name, pv2.version FROM package_links  
INNER JOIN package_versions as pv1
ON parent_package_id = pv1.id
INNER JOIN package_versions as pv2
ON child_package_id = pv2.id
WHERE pv1.name = '@hapi/code' and pv1.version = '8.0.1';

    name    | version 
------------+---------
 @hapi/hoek | 9.0.3
 @hapi/hoek | 9.0.4
(2 rows)
```
So there _are_ 2 dependencies, but probably should only be one.

**Comment by g-k** [comment]
Yeah, I think this is related to #130

**Comment by g-k** [comment]
Looking at:

* https://registry.npmjs.com/@hapi/code code@8.0.1 has the only non-dev constraint`"@hapi/hoek": "9.x.x"`
* across 54 scans we picked up multiple `@hapi/code` patch versions:

```sql
depobs=> SELECT package_links.id AS link_id, pv2.name, pv2.version FROM package_links                                                                         
INNER JOIN package_versions as pv1
ON parent_package_id = pv1.id
INNER JOIN package_versions as pv2
ON child_package_id = pv2.id
WHERE pv1.name = '@hapi/code' and pv1.version = '8.0.1';
-[ RECORD 1 ]-------
link_id | 17029
name    | @hapi/hoek
version | 9.0.3
-[ RECORD 2 ]-------
link_id | 78251
name    | @hapi/hoek
version | 9.0.4
depobs=> SELECT COUNT(*) FROM package_graphs WHERE link_ids && ARRAY[17029, 78251]; -- graphs with either link && is array overlap
-[ RECORD 1 ]
count | 54
```

with 9.0.3 first picked up on 2020-03-04 18:41:34.873305 and 9.0.4 on 2020-04-01 19:06:37.050753:

```sql
depobs=> SELECT inserted_at FROM package_graphs WHERE link_ids @> ARRAY[78251] ORDER BY inserted_at DESC LIMIT 1; 
-[ RECORD 1 ]---------------------------
inserted_at | 2020-04-01 19:06:37.050753

depobs=> SELECT inserted_at FROM package_graphs WHERE link_ids @> ARRAY[17029] ORDER BY inserted_at DESC LIMIT 1; 
-[ RECORD 1 ]---------------------------
inserted_at | 2020-03-04 18:41:34.873305
```

which is consistent with the hapi/hoek publication times:

```
    "created": "2019-03-27T00:33:42.297Z",
    "6.2.0": "2019-03-27T00:33:42.563Z",
    "modified": "2020-03-12T05:12:32.480Z",
...
    "9.0.2": "2020-01-09T18:27:28.494Z",
    "9.0.3": "2020-02-08T06:37:39.724Z",
    "8.5.1": "2020-02-08T06:45:41.173Z",
    "9.0.4": "2020-03-12T05:12:30.107Z"
```

**Comment by g-k** [comment]
relevant graphs:

* https://depobs.dev.mozaws.net/graphs/2969 (for hoek@9.0.3 NB: also includes a hoek@8.5.1)
* https://depobs.dev.mozaws.net/graphs/3038 (for hoek@9.0.4)

**Comment by g-k** [comment]
fixed in #137 

https://depobs.dev.mozaws.net/?manager=npm&package=%40hapi%2Fcode&version=8.0.1 now shows `@hapi/hoek@9.0.4` as a direct dep
